### PR TITLE
add lister generator and refactor common code

### PR DESCRIPTION
**Pull request by varshaprasad96**
Add lister gen, and refactor some common code which was written for client generator.

TODO:

- [ ] Add sample testdata

### CODE REVIEW

**Comment by varshaprasad96** [comment]
Blocked by https://github.com/kcp-dev/apimachinery/pull/21. The template needs to be update with this import.

**Comment by varshaprasad96** [comment]
cc: @fabianvf @njhale

File: pkg/generators/listergen/gen.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+			formattedBytes, err := format.Source(outBytes)
+			if err != nil {
+				root.AddError(err)
+			} else {
+				outBytes = formattedBytes
+			}
+
+			filename := gv.Group.PackageName() + string(version.Version) + util.ExtensionGo
+
+			// TODO: figure out which output file to write to and its path. Should it be appended
```
Comment (ncdc) [inline pkg/generators/listergen/gen.go:204]: ?

**Comment by njhale** [comment]
Superseded by #13

**Pull request closed without merging**